tests/lib/rules/no-node-access.test.ts

```diff
@@ -0,0 +1,202 @@
+import { createRuleTester } from '../test-utils';
+import rule, { RULE_NAME } from '../../../lib/rules/no-node-access';
+
+const ruleTester = createRuleTester({
+  ecmaFeatures: {
+    jsx: true,
+  },
+});
+
+ruleTester.run(RULE_NAME, rule, {
+  valid: [
+    {
+      code: `
+        const buttonText = screen.getByText('submit');
+      `,
+    },
+    {
+      code: `
+        const buttonText = document.firstChild;
+        expect(buttonText.textContent).toBe('submit');
+      `,
+    },
+    {
+      // custom objects with the same properties names as DOM properties are allowed
+      code: `
+        const myObj = {
+          method: () => ({
+            firstChild: 'Some custom text'
+          }),
+        };
+        const text = myObj.method().firstChild;
+      `,
+    },
+  ],
+  invalid: [
+    {
+      code: `document.getElementById('submit-btn').closest('button');`,
+      errors: [
+        {
+          messageId: 'noNodeAccess',
+        },
+      ],
+    },
+    {
+      code: `
+        const closestButton = document.getElementById('submit-btn').closest('button');
+        expect(closestButton).toBeInTheDocument();
+      `,
+      errors: [
+        {
+          messageId: 'noNodeAccess',
+        },
+      ],
+    },
+    {
+      code: `screen.getByText('submit').closest('button');`,
+      errors: [
+        {
+          line: 1,
+          column: 28,
+          messageId: 'noNodeAccess',
+        },
+      ],
+    },
+    {
+      code: `
+        expect(screen.getByText('submit').closest('button').textContent).toBe('Submit');
+      `,
+      errors: [
+        {
+          messageId: 'noNodeAccess',
+        },
+      ],
+    },
+    {
+      code: `
+        const buttons = screen.getAllByRole('button');
+        const buttonA = buttons[1];
```

Comment:
Makes sense. Since this rule is already covering methods from `document` that return an HTMLCollection such as `document.getElementsBy...` only arrays returned from Testing Library methods will be allowed anyway, so I can just remove this check.